crates/sui-json-rpc/src/coin_api.rs

```diff
@@ -219,7 +219,9 @@ impl CoinReadApiServer for CoinReadApi {
         with_tracing!(async move {
             let coin_struct = parse_to_struct_tag(&coin_type)?;
             Ok(if GAS::is_gas(&coin_struct) {
-                Supply { value: 0 }
+                Supply {
+                    value: TOTAL_SUPPLY_SUI,
```

Comment:
This likely needs to be in Mist

Reply:
Oops on indexer v2 we use TOTAL_SUPPLY_SUI too so that should also be changed, to be aligned with other currencies, where the value in the supply object is directly returned.

Reply:
good callout, I copied the logic from indexer v2, will address.